class.jetpack-jitm.php

```diff
@@ -233,14 +233,10 @@ static function get_messages( $message_path, $query ) {
 
 		// attempt to get from cache
 		$envelopes  = get_transient( 'jetpack_jitm_' . $path );
-		$from_cache = false;
 
-		// if something is in the cache and it was put in the cache after the last heartbeat or sync, use it
-		$last_heartbeat = Jetpack_Options::get_option( 'last_heartbeat', time() );
-		$last_sync      = Jetpack_Options::get_option( 'last_sync', time() );
-		if ( $envelopes && $last_heartbeat < $envelopes['response_time'] || $last_sync < $envelopes['response_time'] ) {
-			$from_cache = true;
-		}
+		// if something is in the cache and it was put in the cache after the last sync we care about, use it
+		$last_sync = (int) get_transient( 'jetpack_last_sync' );
+		$from_cache = $envelopes && $last_sync < $envelopes[ 'response_time' ];
```

Comment:
It occurs to me that 'response_time' isn't necessarily the best name for this envelope value, maybe `expires_at`?

Furthermore, your expiry for `jetpack_last_sync` is 1 hour. This means after 1 hour, a synced plugin change is no longer going to bust the cache.

So, gaming this out, if a user activates a plugin, goes away for 61 minutes, then comes back to a wp-admin page, they will not see an updated JITM on that page. That doesn't seem like acceptable behaviour to me. Is there any particular reason why `jetpack_last_sync` needs to be a transient?